Backend/app/core/tts.py

```
@@ -0,0 +1,307 @@
+
+    def _configure_character_voice(self, voice_config: Dict):
+        try:
+            voices = self.engine.getProperty('voices')
+            
+            if voices and len(voices) > voice_config["voice_index"]:
+                selected_voice = voices[voice_config["voice_index"]]
+                self.engine.setProperty('voice', selected_voice.id)
+                logger.debug(f"🎯 Using voice: {selected_voice.name if hasattr(selected_voice, 'name') else 'Default'}")
+            elif voices:
+                self.engine.setProperty('voice', voices[0].id)
+                logger.debug(f"🎯 Fallback to first voice")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)